### PR TITLE
Update Github action workflows

### PR DESCRIPTION
**Pull request by tylervz**
Update the GitHub actions workflows to use the latest version of multiple actions, including `actions/checkout`,  `actions/setup-java`, `actions/cache`, and `release-drafter/release-drafter`:

https://github.com/actions/checkout/tags
https://github.com/actions/setup-java/tags
https://github.com/actions/cache?tab=readme-ov-file#v4
https://github.com/release-drafter/release-drafter/tags

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -10,48 +10,58 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
Comment (matrei) [inline .github/workflows/gradle.yml]: We are trying to switch to `liberica` across the board.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -10,48 +10,58 @@ jobs:
...
           java-version: 17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Comment (matrei) [inline .github/workflows/gradle.yml]: Use v4

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -10,48 +10,58 @@ jobs:
...
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
```
Comment (matrei) [inline .github/workflows/gradle.yml]: We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -10,48 +10,58 @@ jobs:
...
+          build-scan-terms-of-use-agree: "yes"
       - name: Run Build
         id: build
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew build
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (matrei) [inline .github/workflows/gradle.yml]: `DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

File: .github/workflows/gradle.yml [hunk 5]
```diff
@@ -10,48 +10,58 @@ jobs:
...
     needs: ["build"]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
Comment (matrei) [inline .github/workflows/gradle.yml]: Use `liberica` instead of `temurin`

File: .github/workflows/gradle.yml [hunk 6]
```diff
@@ -10,48 +10,58 @@ jobs:
...
           java-version: '17'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Comment (matrei) [inline .github/workflows/gradle.yml]: Use v4

File: .github/workflows/gradle.yml [hunk 7]
```diff
@@ -10,48 +10,58 @@ jobs:
...
+        with:
+          build-scan-publish: true
```
Comment (matrei) [inline .github/workflows/gradle.yml]: We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

File: .github/workflows/gradle.yml [hunk 8]
```diff
@@ -10,48 +10,58 @@ jobs:
...
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
       - name: Publish to Artifactory (repo.grails.org)
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew -Dorg.gradle.internal.publish.checksums.insecure=true publish
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (matrei) [inline .github/workflows/gradle.yml]: `DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -13,11 +13,11 @@ jobs:
...
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           token: ${{ secrets.GH_TOKEN }}
       - name: Set up JDK
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
Comment (matrei) [inline .github/workflows/release.yml]: Use `liberica` instead of `temurin`

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -35,12 +35,18 @@ jobs:
       - name: Set the current release version
         id: release_version
         run: echo "value=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Comment (matrei) [inline .github/workflows/release.yml]: Use v4

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -35,12 +35,18 @@ jobs:
...
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
```
Comment (matrei) [inline .github/workflows/release.yml]: We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

File: .github/workflows/release.yml [hunk 12]
```diff
@@ -35,12 +35,18 @@ jobs:
...
+          build-scan-terms-of-use-agree: "yes"
       - name: Run Assemble
         if: success()
         id: assemble
-        uses: gradle/gradle-build-action@v2
-        with:
-          arguments: assemble
+        run: ./gradlew assemble
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (matrei) [inline .github/workflows/release.yml]: `DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

File: .github/workflows/release.yml [hunk 13]
```diff
@@ -57,7 +63,10 @@ jobs:
         run: echo $SECRING_FILE | base64 -d > ${{ github.workspace }}/secring.gpg
       - name: Publish to Sonatype OSSRH
         id: publish
-        uses: gradle/gradle-build-action@v2
+        run: >
+          ./gradlew -Psigning.secretKeyRingFile=${{ github.workspace }}/secring.gpg
+          publishToSonatype 
+          closeSonatypeStagingRepository
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (matrei) [inline .github/workflows/release.yml]: `DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

File: .github/workflows/release.yml [hunk 14]
```diff
@@ -84,19 +88,23 @@ jobs:
         with:
           token: ${{ secrets.GH_TOKEN }}
           ref: v${{ needs.publish.outputs.release_version }}
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Comment (matrei) [inline .github/workflows/release.yml]: Use v4

File: .github/workflows/release.yml [hunk 15]
```diff
@@ -84,19 +88,23 @@ jobs:
...
+        with:
+          build-scan-publish: true
```
Comment (matrei) [inline .github/workflows/release.yml]: We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

File: .github/workflows/release.yml [hunk 16]
```diff
@@ -84,19 +88,23 @@ jobs:
...
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
       - name: Nexus Staging Close And Release
-        uses: gradle/gradle-build-action@v2
+        run: >
+          ./gradlew findSonatypeStagingRepository
+          releaseSonatypeStagingRepository 
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (matrei) [inline .github/workflows/release.yml]: `DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

**Review by matrei — CHANGES_REQUESTED** [review]

**Comment by tylervz** [comment]
@matrei I updated all the `gradle/actions/setup-gradle` to v4 and declared them with
```
- name: Setup Gradle
        uses: gradle/actions/setup-gradle@v4
        with:
          develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
And then I removed the `DEVELOCITY_ACCESS_KEY` environment variable declarations.

**Review by matrei — APPROVED** [review]

**Comment by matrei** [comment]
@tylervz  Thanks!, looking good :star:

**Pull request merged**